modules/third-party/biz-aQute-bnd-annotation/src/main/java/aQute/bnd/annotation/metatype/Configurable.java

```diff
@@ -278,6 +278,8 @@ Object convertArray(Type componentType, Object o) throws Exception {
 					return s.getBytes(UTF_8);
 				if (componentType == Character.class || componentType == char.class)
 					return s.toCharArray();
+				if (s.isEmpty())
```

Comment:
@ericyanLr You can not just simply return an empty String array here, as the componentType might not be String. You should modify "toCollection", make it return an empty String array when the input is blank, then it will go through the type processing and creating arrays with right types.

Reply:
@tinatian , thanks for catching this and the guidance, that makes sense to me. I'll make the changes and re-send.